libwallet/instantswap.go

```diff
@@ -126,47 +126,57 @@ func (mgr *AssetsManager) StartScheduler(ctx context.Context, params instantswap
-		// Current rate source supported Binance and Bittrex always returns
-		// ticker.LastTradePrice in's the quote asset unit e.g DCR-BTC, LTC-BTC.
-		// We will also do this when and if USDT is supported.
-		if strings.EqualFold(fromCur, "btc") {
-			rateSourceRate = 1 / ticker.LastTradePrice
+		// set the max send amount to the max limit set by the server
+		invoicedAmount := res.Max
+		walletBalance := sourceAccountBalance.Spendable.ToCoin()
+
+		estimatedBalanceAfterExchange := walletBalance - invoicedAmount
+		// if the max send limit is 0, then the server does not have a max limit
+		// constraint so we can send the entire source wallet balance
+		if res.Max == 0 || estimatedBalanceAfterExchange < params.BalanceToMaintain {
+			invoicedAmount = walletBalance - params.BalanceToMaintain // deduct the balance to maintain from the source wallet balance
 		}
 
-		serverRateStr := values.StringF(values.StrServerRate, params.Order.ExchangeServer.Server, fromCur, exchangeServerRate, toCur)
-		log.Info(serverRateStr)
-		binanceRateStr := values.StringF(values.StrCurrencyConverterRate, source, fromCur, rateSourceRate, toCur)
-		log.Info(binanceRateStr)
-
-		// check if the server rate deviates from the market rate by ± 5%
-		// exit if true
-		percentageDiff := math.Abs((exchangeServerRate-rateSourceRate)/((exchangeServerRate+rateSourceRate)/2)) * 100
-		if percentageDiff > params.MaxDeviationRate {
-			log.Error("exchange rate deviates from the market rate by more than 5%")
-			return errors.E(op, "exchange rate deviates from the market rate by more than 5%")
+		if invoicedAmount <= 0 {
+			errMsg := fmt.Errorf("balance to maintain is the same or greater than wallet balance(Current Balance: %v, Balance to Maintain: %v)", walletBalance, params.BalanceToMaintain)
+			log.Error(errMsg)
+			return errors.E(op, errMsg)
 		}
 
-		// set the max send amount to the max limit set by the server
-		invoicedAmount := res.Min
-
-		// if the max send limit is 0, then the server does not have a max limit constraint
-		// so we can send the entire source wallet balance
-		// if res.Max == 0 {
-		// 	invoicedAmount = sourceAccountBalance.Spendable.ToCoin() - params.BalanceToMaintain // deduct the balance to maintain from the source wallet balance
-		// }
-
-		log.Info("Order Scheduler: check balance after exchange")
-		estimatedBalanceAfterExchange := sourceAccountBalance.Spendable.ToCoin() - invoicedAmount
-		if estimatedBalanceAfterExchange < params.BalanceToMaintain {
-			log.Error("source wallet balance after the exchange would be less than the set balance to maintain")
-			return errors.E(op, "source wallet balance after the exchange would be less than the set balance to maintain") // stop scheduling if the source wallet balance after the exchange would be less than the set balance to maintain
+		if invoicedAmount == walletBalance {
+			errMsg := "Specify a little balance to maintain to cover for transaction fees... e.g 0.001 for DCR to BTC or LTC swaps"
```

Comment:
this band aid can work for now, but we would need to improve the logic to automatically do the necessary calculation from our end, similar to what happens when a user clicks "max" on the send page

Reply:
> similar to what happens when a user clicks "max" on the send page


Same thought, but not sure if the max send has been implemented yet.